python_modules/dagster/dagster/_daemon/asset_daemon.py

```diff
@@ -137,11 +138,30 @@ def run_iteration(
 
         evaluations_by_asset_key = {evaluation.asset_key: evaluation for evaluation in evaluations}
 
+        # Look for existing runs from this evaluation ID (in case the tick failed partway through launching runs)
+        runs_filter = RunsFilter(tags={ASSET_EVALUATION_ID_TAG: str(new_cursor.evaluation_id)})
+        existing_runs = instance.get_runs(runs_filter)
+
+        print(str(existing_runs))
+
         for run_request in run_requests:
             yield
 
             asset_keys = check.not_none(run_request.asset_selection)
 
+            asset_key_str = ", ".join([asset_key.to_user_string() for asset_key in asset_keys])
+
+            if any(
+                (run.asset_selection or {}) == set(asset_keys)
```

Comment:
one thing that I am not certain of here is whether this check is sufficient - like if the redone tick could somehow produce a different set of assets that isn't exactly the same and still keeps running in a loop

Reply:
I think the optimal check would be if this asset partition has previously been launched for this evaluation